pkg/operator/status/status_controller.go

```diff
@@ -124,7 +124,10 @@ func (c StatusSyncer) sync() error {
 	}
 	clusterOperatorObj := originalClusterOperatorObj.DeepCopy()
 
-	if detailedSpec.ManagementState == operatorv1.Unmanaged && !management.IsOperatorAlwaysManaged() {
+	// management.IsOperatorAlwaysManaged() means the operator can't be set to unmanaged state.
+	// so if a cluster admin sets an operator as unmanaged when it can't be unmanaged, then we set the
+	// conditions to unknown, as we are now in an indeterminate state
+	if detailedSpec.ManagementState == operatorv1.Unmanaged && management.IsOperatorAlwaysManaged() {
```

Comment:
is the operator is supposed to be always managed, why wouldn't we just stomp their change back to "managed"?

Because we still need to provide a backdoor way for the admin to hack stuff up?  what does "IsOperatorAlwaysManaged" actually reflect/check/mean?